azext_iot/product/params.py

```diff
@@ -0,0 +1,149 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+CLI parameter definitions.
+"""
+
+from azure.cli.core.commands.parameters import get_three_state_flag, get_enum_type
+from azext_iot.product.shared import AuthType, BadgeType, TaskType
+
+
+def load_product_params(self, _):
+    with self.argument_context('iot product') as c:
+        c.argument(
+            'product_name',
+            options_list=['--product-name'],
+            help='Product name to display in catalog',
+            arg_group='IoT Device Certification'
+        )
+        c.argument(
+            'test_id',
+            options_list=['--test-id'],
+            help='The generated Id for the device certification test',
+            arg_group='IoT Device Certification'
+        )
+        c.argument(
+            'badge_type',
+            options_list=['--badge-type', '--bt'],
+            help='The type of certification badge',
+            arg_group='IoT Device Certification',
+            arg_type=get_enum_type(BadgeType)
+        )
+    with self.argument_context('iot product test') as c:
+        c.argument(
+            'working_folder',
+            options_list=['--working-folder', '--wf'],
+            help='The folder to create in current path',
+            arg_group='IoT Device Certification'
+        )
+        c.argument(
+            'auth_type',
+            options_list=['--authentication-type', '--at'],
+            help='How the device will authenticate to testing service Device Provisioning Service',
+            arg_group='IoT Device Certification',
+            arg_type=get_enum_type(AuthType)
+        )
+        c.argument(
+            'provisioning',
+            options_list=['--provisioning'],
+            help='Determines whether the service generates provisioning configuration. '
+            + 'Only applies to SymmetricKey and ConnectionString provisioning types',
+            arg_group='IoT Device Certification'
+        )
+        c.argument(
+            'configuration_file',
+            options_list=['--configuration-file', '--cf'],
+            help='Path to device test JSON file',
+            arg_group='IoT Device Certification'
+        )
+    with self.argument_context('iot product test search') as c:
+        c.argument(
+            'product_id',
+            options_list=['--product-id', '--pi'],
```

Comment:
if possible, it would be good to have any "id" related arguments be single chars